indexer/bin/indexer.js

```diff
@@ -6,6 +6,93 @@ export default {
    * @returns {Promise<Response>}
    */
   async fetch(request, env, ctx) {
-    return new Response('Not Implemented', { status: 501 })
+    if (request.method !== 'POST') {
+      return new Response('Method Not Allowed', { status: 405 })
+    }
+    const payload = await request.json()
+    const pathname = new URL(request.url).pathname
+    if (pathname === '/proof-set-created') {
+      if (!payload.set_id || !payload.owner) {
+        return new Response('Bad Request', { status: 400 })
+      }
+      await env.DB.prepare(
```

Comment:
I think we should move request handling logic into handlers similar to what we did in projects like `spark-api`. I think it'll make a great difference when maintaining this project as this already gets a bit tought to read.

Reply:
Let's not worry about maintainability for now, this can be refactored post launch

Reply:
I was thinking about the same - we should extract route handlers into individual functions, and it's a refactoring to do post M1.